tests/UI/Base.php

```diff
@@ -62,8 +63,23 @@ public function createId() {
 	public function addOnLoadCode($code) {
 		$this->on_load_code[] = $code;
 	}
+	public function getOnLoadCodeAsync() {
+	}
+}
+
+class IncrementalSignalGenerator extends \ILIAS\UI\Implementation\Component\SignalGenerator {
+
+	protected $id = 0;
+
+	protected function createId() {
+		return 'signal_' . ++$this->id;
+	}
 }
 
+class SignalGeneratorMock extends \ILIAS\UI\Implementation\Component\SignalGenerator {}
```

Comment:
Yes, currently the unit tests for the modals make use of this class when testing getting and resetting their signals.